common/cache/cache_types.go

```diff
@@ -0,0 +1,25 @@
+package cache
+
+import (
+	"container/list"
+	"sync"
+)
+
+// LRUCache thread safe fixed size LRU cache
+type LRUCache struct {
+	lru *LRU
+	m   sync.RWMutex
```

Comment:
We don't use RLock/RUnlock so a sync.Mutex could be substituted here

Reply:
Ah yeah was originally looking at doing read locks for some of the calls (Get & Contains) but turns out didn't really fit in the end swapped back to normal mutex